hazelcast/src/main/java/com/hazelcast/util/BufferingInputStream.java

```diff
@@ -0,0 +1,81 @@
+/*
```

Comment:
I just added `readFullyOrNothing()` and `readFully()` to `IOUtil`, so we can re-use all this code for the Near Cache pre-loader and HotRestart. Maybe I'll pull that part out, since the code is tested separately and to shrink this PR.